src/steamship_langchain/llms/openai.py

```diff
@@ -1,63 +1,106 @@
 import hashlib
+import json
 import logging
-from typing import Any, List, Mapping, Optional
+from collections import defaultdict
+from typing import Any, Dict, Generator, List, Mapping, Optional
 
-import tiktoken
-from langchain.llms.base import BaseLLM, Generation, LLMResult
-from pydantic import Field
+from langchain.llms.base import Generation, LLMResult
+from langchain.llms.openai import BaseOpenAI
+from pydantic import root_validator
 from steamship import Block, File, Steamship, SteamshipError
 from steamship.data import TagKind, TagValueKey
 
-
-class OpenAI(BaseLLM):
+PLUGIN_HANDLE: str = "openai"
+ARGUMENT_WHITELIST = {
+    "client",
+    "model_name",
+    "temperature",
+    "max_tokens",
+    "top_p",
+    "frequency_penalty",
+    "presence_penalty",
+    "n",
+    "best_of",
+    "model_kwargs",
+    "openai_api_key",
+    "batch_size",
+    "request_timeout",
+    # "logit_bias",
+    "max_retries",
+    "callback_manager",
+    "cache",
+    "verbose",
+}
+
+
+class OpenAI(BaseOpenAI):
     """Implements LangChain LLM interface in a Steamship-compatible fashion, allowing use in chains/agents as required.
 
     NOTE: This provides a **synchronous** interaction with the LLM backend.
     """
 
-    client: Steamship
-    plugin_handle: str = Field(default="gpt-3", const=True)
-    batch_size: int = Field(default=20)
-    temperature: float = 0.8
-    max_words: int = 256
-    n: int = 1
-    best_of: int = 1
-
-    @property
-    def _llm_type(self) -> str:
-        return f"steamship-{self.plugin_handle}"
+    client: Steamship  # We can use validate_environment to add the client here
 
     @property
     def _identifying_params(self) -> Mapping[str, Any]:
         return {
-            "plugin_handle": self.plugin_handle,
+            "plugin_handle": PLUGIN_HANDLE,
             "workspace_handle": self.client.get_workspace().handle,
+            **super()._identifying_params,
+        }
+
+    def _invocation_params(self, stop: Optional[List[str]] = None):
+        stop_str = ",".join(stop) if stop is not None else ""
+
+        return {
+            **self._default_params,
+            "stop": stop_str,
+        }
+
+    @property
+    def _default_params(self) -> Dict[str, Any]:
+        """Get the default parameters for calling Steamship's OpenAI Plugin."""
+        normal_params = {
+            "model_name": self.model_name,
             "temperature": self.temperature,
-            "max_words": self.max_words,
-            "cache": self.cache,
+            "max_tokens": self.max_tokens,
+            "top_p": self.top_p,
+            "frequency_penalty": self.frequency_penalty,
+            "presence_penalty": self.presence_penalty,
             "n": self.n,
             "best_of": self.best_of,
+            "request_timeout": self.request_timeout,
+            "max_retries": self.max_retries,
+            # "logit_bias": self.logit_bias,
         }
 
-    def get_num_tokens(self, text: str) -> int:
```

Comment:
removing this _should_ be fine, given that Steamship operates in 3.8.1 envs, but we should be sure to validate with some deployments to be sure.